test/cpp/end2end/xds/xds_end2end_test_lib.cc

```diff
@@ -1014,7 +1014,9 @@ absl::optional<AdsServiceImpl::ResponseState> XdsEnd2endTest::WaitForNack(
            response_state->state != AdsServiceImpl::ResponseState::NACKED;
   };
   do {
-    const Status status = SendRpc();
+    // Some tests for example the XdsSecurityTests need a higher timeout from
+    // the default 1000ms for some builds.
+    const Status status = SendRpc(RpcOptions().set_timeout_ms(5000));
```

Comment:
Please add an optional parameter to this method to pass in the `RpcOptions` and then override it only in the affected tests, rather than hard-coding this default for all tests.

Reply:
I don't mind doing that but, I couldn't think of a reason where a `WaitForNack()` test would expect a Deadline Exceeded

Reply:
but from a futuristic perspective, if some test ends up requiring a RpcOptions() then yes, it does make sense to add that argument now itself.

Reply:
Done!